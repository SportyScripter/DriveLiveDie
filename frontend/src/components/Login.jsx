import { useForm } from "react-hook-form";

export function Login() {
  const { register, handleSubmit } = useForm();

  const onSubmit = (data) => {
    fetch("http://localhost:8008/auth/login", {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json",
      },
    }).then((response) => {
      if (response.ok) {
        response.json().then((data) => {
          localStorage.setItem("access_token", data.token);
          window.location.reload();
        });
      }
    });
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>
        Email
        <input {...register("email")} />
      </label>
      <label>
        Password
        <input {...register("password")} />
      </label>
      <button type="submit">Login</button>
    </form>
  );
}
