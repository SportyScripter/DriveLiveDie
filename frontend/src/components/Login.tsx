import { useState } from "react";
import { useForm } from "react-hook-form";

export function Login() {
  const [error, setError] = useState("");
  const { formState, register, handleSubmit } = useForm();

  const touched = Object.keys(formState.touchedFields).length === 2;

  const onSubmit = (data: any) => {
    fetch("http://localhost:8008/auth/login", {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json",
      },
    }).then((response) => {
      if (response.ok) {
        response.json().then((data) => {
          localStorage.setItem("access_token", data.token);
          window.location.href = "/dashboard";
        });
      } else {
        console.log(response);
        setError("Nieprawidłowa nazwa użytkownika lub hasło.");
      }
    });
  };

  return (
    <form
      className="bg-white px-8 pt-6 pb-8 mb-4"
      onSubmit={handleSubmit(onSubmit)}
    >
      {error && (
        <div className="error mt-3 mb-3 bg-red-100 border-red-300 p-2">
          {error}
        </div>
      )}
      <div className="mb-4">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="email"
        >
          Adres e-mail
        </label>
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="email"
          type="text"
          placeholder="email"
          {...register("email")}
        />
      </div>
      <div className="mb-6">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="password"
        >
          Hasło
        </label>
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
          id="password"
          type="password"
          placeholder="******************"
          {...register("password")}
        />
      </div>
      <div className="flex items-center justify-between">
        <button
          disabled={!touched}
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
          type="submit"
        >
          Zaloguj
        </button>
      </div>
    </form>
  );
}
