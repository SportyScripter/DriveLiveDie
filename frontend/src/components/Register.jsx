import { useForm } from "react-hook-form";

export function Register() {
  const { register, handleSubmit } = useForm();

  const onSubmit = (data) => {
    console.log(data);

    fetch("http://localhost:8008/auth/create-user", {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json",
      },
    });
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>
        Name
        <input {...register("name")} />
      </label>
      <label>
        last name
        <input {...register("last_name")} />
      </label>
      <label>
        username
        <input {...register("username")} />
      </label>
      <label>
        Email
        <input {...register("email")} />
      </label>
      <label>
        Password
        <input {...register("password")} />
      </label>
      <button type="submit">Register</button>
    </form>
  );
}
