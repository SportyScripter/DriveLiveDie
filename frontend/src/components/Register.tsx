import { useState } from "react";
import { useForm } from "react-hook-form";

export function Register() {
  const { formState, register, handleSubmit } = useForm();

  const [error, setError] = useState("");

  const touched = Object.keys(formState.touchedFields).length === 5;

  const onSubmit = (data: any) => {
    console.log(data);

    fetch("http://localhost:8008/auth/register", {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json",
      },
    }).then((response) => {
      if (response.ok) {
        response.json().then((data) => {
          localStorage.setItem("access_token", data.token);
          window.location.href = "/dashboard";
        });
      } else {
        console.log(response);
        setError("Błędne dane.");
      }
    });
  };

  return (
    <form
      className="bg-white px-8 pt-6 pb-8 mb-4"
      onSubmit={handleSubmit(onSubmit)}
    >
      {error && (
        <div className="error mt-3 mb-3 bg-red-100 border-red-300 p-2">
          {error}
        </div>
      )}
      <div className="mb-4">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="username"
        >
          Nazwa użytkownika
        </label>
        <input
          {...register("username")}
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="username"
          type="text"
          placeholder="Nazwa użytkownika"
        />
      </div>
      <div className="mb-4">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="first_name"
        >
          Imię
        </label>
        <input
          {...register("first_name")}
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="first_name"
          type="text"
          placeholder="Imię"
        />
      </div>
      <div className="mb-4">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="last_name"
        >
          Nazwisko
        </label>
        <input
          {...register("last_name")}
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="last_name"
          type="text"
          placeholder="Nazwisko"
        />
      </div>
      <div className="mb-4">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="email"
        >
          Adres e-mail
        </label>
        <input
          {...register("email")}
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
          id="email"
          type="text"
          placeholder="email"
        />
      </div>
      <div className="mb-6">
        <label
          className="block text-gray-700 text-sm font-bold mb-2"
          htmlFor="password"
        >
          Hasło
        </label>
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
          id="password"
          type="password"
          placeholder="******************"
          {...register("password")}
        />
      </div>
      <div className="flex items-center justify-between">
        <button
          disabled={!touched}
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
          type="submit"
        >
          Zarejestruj
        </button>
      </div>
    </form>
  );
}
