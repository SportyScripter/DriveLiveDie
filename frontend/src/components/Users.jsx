import React from "react";

export function Users() {
  const [users, setUsers] = React.useState([]);

  React.useEffect(() => {
    const fetchUsers = async () => {
      const response = await fetch("http://localhost:8008/users");
      const data = await response.json();
      if (response.ok) {
        setUsers(data);
      }
    };

    fetchUsers();
  }, []);

  return (
    <div>
      <h1>Users</h1>

      <ul>
        {users.map((user) => (
          <li key={user.id}>{user.email}</li>
        ))}
      </ul>
    </div>
  );
}
