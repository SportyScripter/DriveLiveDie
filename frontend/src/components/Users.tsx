import React from "react";

interface User {
  id: number;
  email: string;
  username: string;
  first_name: string;
  last_name: string;
}

export function Users() {
  const [users, setUsers] = React.useState<User[]>([]);

  React.useEffect(() => {
    const fetchUsers = async () => {
      const response = await fetch("http://localhost:8008/users");
      const data = await response.json();
      if (response.ok) {
        setUsers(data);
      }
    };

    fetchUsers();
  }, []);

  return (
    <div>
      <h1>Users</h1>

      <ul>
        {users.map((user) => (
          <li key={user.id}>{user.email}</li>
        ))}
      </ul>
    </div>
  );
}
