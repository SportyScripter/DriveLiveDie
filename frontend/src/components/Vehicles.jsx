import React from "react";

export function Vehicles() {
  const [vehicles, setVehicles] = React.useState([]);

  React.useEffect(() => {
    const fetchVehicles = async () => {
      const response = await fetch("http://localhost:8008/vehicles");
      const data = await response.json();
      if (response.ok) {
        setVehicles(data);
      }
    };

    fetchVehicles();
  }, []);

  return (
    <div>
      <h1>Vehicles</h1>

      <ul>
        {vehicles.map((vehicle) => (
          <li key={vehicle.id}>{vehicle.name}</li>
        ))}
      </ul>
    </div>
  );
}
