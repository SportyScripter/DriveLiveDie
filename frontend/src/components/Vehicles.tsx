import React from "react";

interface Vehicle {
  id: number;
  name: string;
}

export function Vehicles() {
  const [vehicles, setVehicles] = React.useState<Vehicle[]>([]);

  React.useEffect(() => {
    const fetchVehicles = async () => {
      const response = await fetch("http://localhost:8008/vehicles");
      const data = await response.json();
      if (response.ok) {
        setVehicles(data);
      }
    };

    fetchVehicles();
  }, []);

  return (
    <div>
      <h1>Vehicles</h1>

      <ul>
        {vehicles.map((vehicle) => (
          <li key={vehicle.id}>{vehicle.name}</li>
        ))}
      </ul>
    </div>
  );
}
