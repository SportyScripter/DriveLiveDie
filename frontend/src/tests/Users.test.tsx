import { render, screen } from "@testing-library/react";
import { Users } from "../components/Users";

import { server } from "./mocks.ts";

describe("Users", () => {
  beforeAll(() => {
    server.listen();
  });
  afterAll(() => server.close());

  test("Renders the view correctly", async () => {
    render(<Users />);
    screen.findByText("Users");
  });

  test("renders users", async () => {
    render(<Users />);
    await screen.findByText("test@test.pl");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Users />);
    expect(asFragment()).toMatchSnapshot();
  });
});
