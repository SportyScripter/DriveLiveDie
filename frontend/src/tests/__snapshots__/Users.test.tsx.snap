// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Users > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Users
    </h1>
    <ul />
  </div>
</DocumentFragment>
`;
